File: results/dnd-rest-benchmark/lifecycle-runs/20260804T070752Z_claude_claude-sonnet-5_ruby-stdlib/lib/handlers/storage.rb
# frozen_string_literal: true

require_relative '../database'

module Handlers
  # Introspection and reset for the persistence layer. `reset` drops and
  # recreates every table, wiping all data (used by the test suite to get a
  # clean slate between scenarios).
  module Storage
    module_function

    def status(_body)
      [200, { driver: 'sqlite', schema_version: Database::SCHEMA_VERSION, initialized: File.exist?(Database.path) }]
    end

    def reset(_body)
      Database.reset_schema
      [200, { ok: true, schema_version: Database::SCHEMA_VERSION }]
    end
  end
end

File: results/dnd-rest-benchmark/lifecycle-runs/20260731T063957Z_claude_claude-sonnet-5_javascript-node/routes/storage.js
import { sendJson } from '../lib/http.js';
import { SCHEMA_VERSION, dbState, resetSchema } from '../lib/db.js';

export function registerStorageRoutes(router) {
  router.get('/health', (req, res) => {
    sendJson(res, 200, { ok: true });
  });

  router.get('/v1/storage/status', (req, res) => {
    sendJson(res, 200, {
      driver: 'sqlite',
      schema_version: SCHEMA_VERSION,
      initialized: dbState.initialized,
    });
  });

  router.post('/v1/storage/reset', (req, res) => {
    resetSchema();
    sendJson(res, 200, { ok: true, schema_version: SCHEMA_VERSION });
  });
}
